### PR TITLE
[FSDE-1009][FSDE-1010][FSDE-283] Improved position of nested rows of Data Table

### PR DESCRIPTION
**Pull request by janibolkvadze**
### Commit Summary

- [x] Added support of proper positioning of 3 levels of rows
- [x] Added support of default expansion of specific rows
- [x] Improved Sorting event emission to provide more information `{ column: string; direction: 'ascending' | 'descending'; data: DataTableRow[] | undefined }` (added a respective interface, `data` is undefined when using Server Side Pagination)
- [x] Improved Page Change event emission to provide more information `{ page: number; data: DataTableRow[] | undefined }` (added a respective interface, `data` is undefined when using Server Side Pagination)
- [x] Changed Selection / Expansion columns order

@SergioQCL @dani-cl-madrid @aryaroudi @evansito

https://user-images.githubusercontent.com/25136131/123306528-b3ac9080-d521-11eb-88b3-48c3ded148a4.mov

### CODE REVIEW

**Comment by SCMAUTO** [comment]
The CI pipeline run successfully in https://jenkinsprod.corp.intranet:8443/job/UX-CHI/job/Productive/job/Chi/job/PR-83/1/. :white_check_mark:

**Comment by SCMAUTO** [comment]
You can check this PRs instance in https://ffa501f0173abba98f9476d13fde468ee1ac8e93-chi-pr.enterprise.ctl.io

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 1]
```diff
@@ -873,10 +911,16 @@ export default class DataTable extends Vue {
       if (columnName && sortIcon) {
         sortIcon.className = `${ICON_CLASS} -xs ${DATA_TABLE_SORT_ICONS.ARROW}`;
         if (currentSort === 'ascending' && this._sortConfig && this._sortConfig.key === columnName) {
+          const sortingData: DataTableSorting = {
+            column: columnName,
+            direction: 'descending',
```
Comment (SergioQCL) [inline src/chi-vue/src/components/data-table/DataTable.tsx:916]: Shouldn't this be ascending?

File: src/chi-vue/src/components/data-table/DataTable.tsx [hunk 2]
```diff
@@ -899,9 +943,15 @@ export default class DataTable extends Vue {
           columnHeadCell?.classList.remove(ACTIVE_CLASS);
           this._sortConfig = undefined;
         } else {
+          const sortingData = {
+            column: columnName,
+            direction: 'ascending',
```
Comment (SergioQCL) [inline src/chi-vue/src/components/data-table/DataTable.tsx:948]: And this descending?

File: src/chi-vue/src/components/data-table/DataTable.tsx (hunk 1 above)
Comment (janibolkvadze) [reply src/chi-vue/src/components/data-table/DataTable.tsx:916]: Nope, in this scope `currentSort === 'ascending'` means that the user has already applied ascending sorting to the column and the next one should switch direction.

Here is a demo:
https://user-images.githubusercontent.com/25136131/123399730-417d8f80-d5a5-11eb-8e55-a98b12495324.mov

Comment (SergioQCL) [reply src/chi-vue/src/components/data-table/DataTable.tsx:916]: Got it, thanks!

**Review by SergioQCL — APPROVED** [review]
LGTM

**Review by dani-cl-madrid — APPROVED** [review]
`ascending` and `descending` should be a type so it can be exported in the future. But it is not a reason for a change request on this PR, so I approve it. Just keep it in mind for future releases.

**Review by aryaroudi — APPROVED** [review]

**Pull request merged**